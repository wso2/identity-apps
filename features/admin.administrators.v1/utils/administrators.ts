/**
 * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import {
    UserBasicInterface,
    UserRoleInterface
} from "@wso2is/admin.core.v1";
import { administratorConfig } from "@wso2is/admin.extensions.v1/configs/administrator";
import { SCIMConfigs } from "@wso2is/admin.extensions.v1/configs/scim";
import { UserAccountTypes } from "@wso2is/admin.users.v1/constants/user-management-constants";

/**
 * Checks whether administrator role is present in the user.
 * @param user - user to check if the user is an admin
 * @returns boolean - true if the user is an admin.
 */
export const isAdminUser = (user: UserBasicInterface): boolean => {
    if (!user?.roles) {
        return false;
    }

    return user.roles.some((role: UserRoleInterface) =>
        role.display === administratorConfig.adminRoleName
    );
};

/**
 * Checks whether the user is the owner of the tenant.
 * @param user - user to check if the user is the owner
 * @returns boolean - true if the user is the owner.
 */
export const isOwner = (user: UserBasicInterface):boolean =>
    user[ SCIMConfigs.scim.enterpriseSchema ]?.userAccountType === UserAccountTypes.OWNER;

/**
 * Checks whether the user is a collaborator.
 * Collaborator users does not have a username with a domain separeated by a `/`.
 * primary/username is NOT a collaborator user.
 *
 * @param user - user to check if the user is a collaborator
 * @returns boolean - true if the user is a collaborator.
 */
export const isCollaboratorUser = (user: UserBasicInterface): boolean =>
    user?.userName?.split("/")?.length === 1;
