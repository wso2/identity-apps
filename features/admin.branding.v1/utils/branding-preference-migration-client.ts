/**
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { BrandingPreferenceAPIResponseInterface } from "@wso2is/common.branding.v1/models";
import cloneDeep from "lodash-es/cloneDeep";
import { CustomTextPreferenceAPIResponseInterface } from "../models/custom-text-preference";

/**
 * Utility class for Branding Preference migrations related operations.
 */
class BrandingPreferenceMigrationClient {
    /**
     * Private constructor to avoid object instantiation from outside
     * the class.
     */
    private constructor() {}

    /**
     * Migrate the theme preference from v1 - v2.
     *
     * @remarks
     * Migrations:
     *     - The `theme` object inside the `preference` doc was introduced later
     *       `colors` & `images` objects were moved inside the respective theme.
     *
     * @param original - Original preference of the user.
     * @param additions - New additions.
     * @returns Migrated branding preference.
     */
    public static migrateCustomTextPreference = (
        original: CustomTextPreferenceAPIResponseInterface,
        brandingPreference: BrandingPreferenceAPIResponseInterface
    ): CustomTextPreferenceAPIResponseInterface => {
        const migratedCustomTextPreference: CustomTextPreferenceAPIResponseInterface = cloneDeep(original);

        // v3 - v4 Migrations!
        if (migratedCustomTextPreference?.preference?.text) {
            if (brandingPreference?.preference?.organizationDetails?.copyrightText) {
                if (migratedCustomTextPreference?.preference?.text["copyright"]) {
                    migratedCustomTextPreference.preference.text[
                        "copyright"
                    ] = brandingPreference.preference.organizationDetails.copyrightText;
                }
            }

            if (brandingPreference?.preference?.organizationDetails?.siteTitle) {
                if (migratedCustomTextPreference?.preference?.text["site.title"]) {
                    migratedCustomTextPreference.preference.text[
                        "site.title"
                    ] = brandingPreference.preference.organizationDetails.siteTitle;
                }
            }
        }

        return migratedCustomTextPreference;
    };
}

export default BrandingPreferenceMigrationClient;
