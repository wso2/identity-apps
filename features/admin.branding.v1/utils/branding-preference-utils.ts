/**
 * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import {
    BrandingPreferenceInterface,
    BrandingPreferenceLayoutInterface,
    BrandingPreferenceThemeInterface,
    DynamicBrandingPreferenceLayoutInterface,
    PredefinedThemes
} from "@wso2is/common.branding.v1/models";
import { AxiosError } from "axios";
import cloneDeep from "lodash-es/cloneDeep";
import merge from "lodash-es/merge";
import { BrandingPreferencesConstants } from "../constants";
import { BrandingPreferenceMeta, LAYOUTS, PredefinedLayouts } from "../meta";

/**
 * Utility class for Branding Preference related operations.
 */
export class BrandingPreferenceUtils {

    /**
     * Private constructor to avoid object instantiation from outside
     * the class.
     */
    private constructor() { }

    /**
     * Migrate the theme preference from v1 - v2.
     *
     * @remarks
     * Migrations:
     *     - The `theme` object inside the `preference` doc was introduced later
     *       `colors` & `images` objects were moved inside the respective theme.
     *
     * @param original - Original preference of the user.
     * @param additions - New additions.
     * @returns Migrated branding preference.
     */
    public static migrateThemePreference = (original: BrandingPreferenceInterface,
        additions: Partial<BrandingPreferenceInterface>): BrandingPreferenceInterface => {

        const migratedThemePreference: BrandingPreferenceInterface = cloneDeep(original);

        // If the preference is of v1, create a `theme` object and set the defaults.
        if (!original.theme && additions.theme) {
            migratedThemePreference.theme = additions.theme;
        }

        for (const theme in PredefinedThemes) {
            // v2 Migrations!
            // If the preference is of v1, move `images` inside the theme.
            if (original.images) {
                migratedThemePreference.theme[ PredefinedThemes[ theme ] ].images = original.images;
            }

            // v3 Migrations!
            const originalPrimaryColor: string = migratedThemePreference.theme[
                PredefinedThemes[ theme ]
            ].colors.primary;

            if (typeof originalPrimaryColor === "string") {
                delete migratedThemePreference.theme[ PredefinedThemes[ theme ] ].colors.primary;
                migratedThemePreference.theme[ PredefinedThemes[ theme ] ].colors.primary = {
                    main: originalPrimaryColor
                };
            }

            const originalSecondaryColor: string = migratedThemePreference.theme[
                PredefinedThemes[ theme ]
            ].colors.secondary;

            if (typeof originalSecondaryColor === "string") {
                delete migratedThemePreference.theme[ PredefinedThemes[ theme ] ].colors.secondary;
                migratedThemePreference.theme[ PredefinedThemes[ theme ] ].colors.secondary = {
                    main: originalSecondaryColor
                };
            }

            if (migratedThemePreference.theme[PredefinedThemes[ theme ]].page) {
                const originalLoginPageBackgroundColor: string = migratedThemePreference.theme[
                    PredefinedThemes[ theme ]
                ].page?.background?.backgroundColor;

                if (!migratedThemePreference.theme[ PredefinedThemes[ theme ] ].colors?.background
                    && originalLoginPageBackgroundColor) {
                    migratedThemePreference.theme[ PredefinedThemes[ theme ] ].colors.background = {
                        body: {
                            main: originalLoginPageBackgroundColor
                        }
                    };
                }

                const originalPageFontColor: string = migratedThemePreference.theme[
                    PredefinedThemes[ theme ]
                ].page?.font?.color;

                if (!migratedThemePreference.theme[ PredefinedThemes[ theme ] ].colors?.text && originalPageFontColor) {
                    migratedThemePreference.theme[ PredefinedThemes[ theme ] ].colors.text = {
                        primary: originalPageFontColor
                    };
                }

                migratedThemePreference.theme[ PredefinedThemes[ theme ] ].loginPage = migratedThemePreference.theme[
                    PredefinedThemes[ theme ]
                ].page;
                delete migratedThemePreference.theme[ PredefinedThemes[ theme ] ].page;
            }

            migratedThemePreference.theme[ PredefinedThemes[ theme ] ] = merge(
                BrandingPreferenceMeta.getThemes()[ theme ],
                migratedThemePreference.theme[ PredefinedThemes[ theme ] ]
            );
        }

        if ("removeAsgardeoBranding" in migratedThemePreference.configs) {
            migratedThemePreference.configs.removeDefaultBranding =
                migratedThemePreference.configs.removeAsgardeoBranding;
            delete migratedThemePreference.configs.removeAsgardeoBranding;
        }

        return migratedThemePreference;
    };

    /**
     * Migrate the layout preference to add all relevant properties to the layout section.
     * And also this method will takecare of older branding preference object and add layout
     * preferences to those objects.
     *
     * @param original - Original preference of the user.
     * @param additions - New additions.
     * @returns Migrated Layout preference.
     */
    public static migrateLayoutPreference = (original: BrandingPreferenceInterface,
        additions: Partial<BrandingPreferenceInterface>): BrandingPreferenceInterface => {

        const migratedPreference: BrandingPreferenceInterface = cloneDeep(original);
        const migratingObject: BrandingPreferenceLayoutInterface = cloneDeep(additions.layout);
        const defaultProperties: DynamicBrandingPreferenceLayoutInterface = cloneDeep(LAYOUTS);

        if (!original.layout && additions.layout) {
            migratedPreference.layout = migratingObject;
        } else if (original.layout && additions.layout) {
            merge(migratingObject, migratedPreference.layout);

            migratedPreference.layout = migratingObject;
        } else if (original.layout && !additions.layout) {
            const migratedLayoutSection: BrandingPreferenceLayoutInterface
                = merge(defaultProperties, migratedPreference.layout);

            migratedPreference.layout = migratedLayoutSection;
        } else {
            migratedPreference.layout = {
                activeLayout: BrandingPreferencesConstants.DEFAULT_LAYOUT,
                ...LAYOUTS
            };
        }

        return migratedPreference;
    };

    /**
     * Get the default branding preference.
     *
     * @param overrides - Object to merge.
     * @returns Default branding preference.
     */
    public static getDefaultBrandingPreference(
        overrides?: Partial<BrandingPreferenceInterface>): BrandingPreferenceInterface {

        return merge(BrandingPreferencesConstants.DEFAULT_PREFERENCE, overrides);
    }

    /**
     * Fetches the theme variables of the core theme (ATM `asgardeo`) from `@wso2is/theme`.
     * And set them as the defaults for the `LIGHT` theme.
     * Also generate the defaults for other themes.
     *
     * @param appTheme - Theme used by the app from `@wso2is/theme`.
     * @returns Predefined theme preference.
     */
    public static getPredefinedThemePreferences = (
        appTheme: string): Promise<BrandingPreferenceThemeInterface> => {

        return import(`../../themes/${ appTheme }/theme-variables.json`)
            .then((response: Record<string, string>) => {
                return Promise.resolve({
                    activeTheme: BrandingPreferencesConstants.DEFAULT_THEME,
                    ...merge(BrandingPreferenceMeta.getThemes(),
                        {
                            [ PredefinedThemes.LIGHT ]: {
                                buttons: {
                                    externalConnection: {
                                        base: {
                                            background: {
                                                backgroundColor:
                                                    response.loginPortalExternalConnectionButtonBackgroundColor
                                            },
                                            border: {
                                                borderRadius: response.buttonBorderRadius
                                            }
                                        }
                                    },
                                    primary: {
                                        base: {
                                            border: {
                                                borderRadius: response.buttonBorderRadius
                                            }
                                        }
                                    },
                                    secondary: {
                                        base: {
                                            border: {
                                                borderRadius: response.buttonBorderRadius
                                            }
                                        }
                                    }
                                },
                                colors: {
                                    background: {
                                        body: {
                                            main: response.pageBackground
                                        },
                                        surface: {
                                            inverted: response.asgardeoBlack
                                        }
                                    },
                                    illustrations: {
                                        accent1: {
                                            main: response.accentColor1
                                        },
                                        accent2: {
                                            main: response.accentColor2
                                        },
                                        accent3: {
                                            main: response.accentColor3
                                        },
                                        primary: {
                                            main: response.primaryColor
                                        },
                                        secondary: {
                                            main: response.secondaryColor
                                        }
                                    },
                                    primary: {
                                        main: response.primaryColor
                                    },
                                    secondary: {
                                        main: response.secondaryColor
                                    }
                                },
                                inputs: {
                                    base: {
                                        background: {
                                            backgroundColor: response.inputBackground
                                        },
                                        border: {
                                            borderRadius: response.defaultBorderRadius
                                        }
                                    }
                                },
                                loginBox: {
                                    border: {
                                        borderRadius: response.loginPortalLoginCardBorderRadius
                                    }
                                }
                            },
                            [ PredefinedThemes.DARK ]: {
                                buttons: {
                                    externalConnection: {
                                        base: {
                                            border: {
                                                borderRadius: response.buttonBorderRadius
                                            }
                                        }
                                    },
                                    primary: {
                                        base: {
                                            border: {
                                                borderRadius: response.buttonBorderRadius
                                            }
                                        }
                                    },
                                    secondary: {
                                        base: {
                                            border: {
                                                borderRadius: response.buttonBorderRadius
                                            }
                                        }
                                    }
                                },
                                colors: {
                                    illustrations: {
                                        accent1: {
                                            main: response.accentColor1
                                        },
                                        accent2: {
                                            main: response.accentColor2
                                        },
                                        accent3: {
                                            main: response.accentColor3
                                        },
                                        primary: {
                                            main: response.primaryColor
                                        },
                                        secondary: {
                                            main: response.secondaryColor
                                        }
                                    },
                                    primary: {
                                        main: response.primaryColor
                                    },
                                    secondary: {
                                        main: response.secondaryColor
                                    }
                                },
                                inputs: {
                                    base: {
                                        border: {
                                            borderRadius: response.defaultBorderRadius
                                        }
                                    }
                                },
                                loginBox: {
                                    border: {
                                        borderRadius: response.loginPortalLoginCardBorderRadius
                                    }
                                }
                            }
                        }
                    )
                });
            })
            .catch((error: AxiosError) => {
                return Promise.reject(error);
            });
    };

    /**
     * Checks if the preview container size is too small to display the preview properly.
     *
     * @param layout - Current selected layout.
     * @param width - Current preview container width.
     * @returns a boolean value.
     */
    public static isLayoutPreviewTrimmed(layout: PredefinedLayouts, width: number): boolean {
        switch(layout) {
            case PredefinedLayouts.LEFT_ALIGNED:
            case PredefinedLayouts.RIGHT_ALIGNED:
                return width < BrandingPreferencesConstants.BRANDING_PREVIEW_MIN_WIDTH_SIDE_ALIGNED;
            case PredefinedLayouts.LEFT_IMAGE:
            case PredefinedLayouts.RIGHT_IMAGE:
                return width < BrandingPreferencesConstants.BRANDING_PREVIEW_MIN_WIDTH_SIDE_IMAGE;
            default:
                return false;
        }
    }
}
