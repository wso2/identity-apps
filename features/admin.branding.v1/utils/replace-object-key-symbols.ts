/**
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import isObject from "lodash-es/isObject";
import transform from "lodash-es/transform";

/**
 * Replaces all occurrences of a symbol in an object's keys recursively.
 *
 * @example
 * ```
 * // Returns { "foo-bar": "baz" }
 * replaceSymbolsRecursively({ "foo_bar": "baz" }, "_", "-");
 * ```
 * @param obj - Object to replace symbols in.
 * @param targetSymbol - Symbol to replace.
 * @param replacementSymbol - Symbol to replace with.
 * @returns Object with replaced symbols.
 */
const replaceObjectKeySymbols = (
    obj: Record<string, unknown>,
    targetSymbol: string,
    replacementSymbol: string
): Record<string, unknown> => {
    return transform(obj, (result: Record<string, unknown>, value: Record<string, unknown>, key: string) => {
        const newKey: string = key.replaceAll(targetSymbol, replacementSymbol);

        if (isObject(value)) {
            // If the value is an object, recursively call the function
            result[newKey] = replaceObjectKeySymbols(value, targetSymbol, replacementSymbol);
        } else {
            // If not an object, simply assign the value
            result[newKey] = value;
        }
    });
};

export default replaceObjectKeySymbols;
