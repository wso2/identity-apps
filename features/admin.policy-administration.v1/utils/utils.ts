/**
 * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use
 * this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

export const formatXML = (xml: string): string => {
    if (!xml || typeof xml !== "string") {
        console.warn("Invalid XML input for formatting.");
        return xml || "";
    }

    const PADDING: string = " ".repeat(4); // Indentation level
    const regex: RegExp = /(>)(<)(\/*)/g;

    try {
        // Add line breaks between tags
        const formatted = xml.replace(regex, "$1\n$2$3");

        // Detect if already formatted (basic check for line breaks and indentation)
        const alreadyFormatted = formatted.split("\n").some((line) => line.startsWith(PADDING));

        if (alreadyFormatted) {
            return xml; // Return unmodified if already formatted
        }

        let pad = 0;

        return formatted
            .split("\n")
            .map((line: string) => {
                if (line.match(/<\/\w/)) pad -= 1;

                const indent = PADDING.repeat(Math.max(pad, 0));
                const formattedLine = `${indent}${line}`;

                if (line.match(/<\w[^>]*[^/]>.*$/)) pad += 1;

                return formattedLine;
            })
            .join("\n");
    } catch (error) {
        console.error("Error formatting XML:", error);
        return xml; // Return raw XML in case of errors
    }
};

export const unformatXML = (xml: string): string => {
    if (!xml || typeof xml !== "string") {
        console.warn("Invalid XML input for unformatting.");
        return xml || "";
    }

    try {
        // Remove all line breaks, tabs, and excessive spaces
        return xml
            .replace(/>\s+</g, "><") // Remove spaces between tags
            .replace(/\s{2,}/g, " ") // Replace multiple spaces with a single space
            .trim(); // Remove leading/trailing spaces
    } catch (error) {
        console.error("Error unformatting XML:", error);
        return xml; // Return the original XML if unformatting fails
    }
};

