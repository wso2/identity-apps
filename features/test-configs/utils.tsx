/**
 * Copyright (c) 2021-2024, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

// import { AuthProvider } from "@asgardeo/auth-react";
import {  RenderResult, render as rtlRender } from "@testing-library/react";
import { AccessControlProvider } from "@wso2is/access-control";
import React, { PropsWithChildren, ReactElement } from "react";
import { Provider } from "react-redux";
import { mockStore } from "./__mocks__/redux/redux-store";
import ReduxStoreStateMock from "./__mocks__/redux/redux-store-state";
// import { AuthenticateUtils } from "../src/features/authentication/utils/authenticate-utils";
// import { PreLoader } from "../src/features/core/components/pre-loader/pre-loader";

/**
 * Custom render method to includes things like global context providers, data stores, etc.
 * @see {@link https://testing-library.com/docs/react-testing-library/setup#custom-render} for more info.
 *
 * @param ui - Component to render.
 * @param allowedScopes - Set of allowed scopes for the logged in user.
 * @param featureConfig - UI Features configuration i.e permissions etc.
 * @param initialState - Redux store initial state.
 * @param store - Mocked store.
 * @param renderOptions - Render options.
 *
 * @returns RenderResult
 */
const render = (
    ui: ReactElement,
    {
        allowedScopes = "internal_login",
        featureConfig = window[ "AppUtils" ].getConfig().ui.features,
        initialState = ReduxStoreStateMock,
        store = mockStore(initialState),
        ...renderOptions
    }: any = {}
): RenderResult => {

    const Wrapper = (props: PropsWithChildren): ReactElement => {

        const { children } = props;

        return (
        // <AuthProvider
        //     config={ AuthenticateUtils.getInitializeConfig() }
        //     fallback={ <PreLoader /> }
        //     getAuthParams={ AuthenticateUtils.getAuthParams }
        // >
            <Provider store={ store }>
                <AccessControlProvider
                    allowedScopes={ allowedScopes }
                    features={ featureConfig }
                    organizationType="FIRST_LEVEL_ORGANIZATION"
                >
                    { children }
                </AccessControlProvider>
            </Provider>
            // </AuthProvider>
        );
    };

    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
