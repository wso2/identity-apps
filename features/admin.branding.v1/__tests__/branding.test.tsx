/**
 * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { render, screen, waitFor } from "@wso2is/unit-testing/utils";
import { MotionGlobalConfig } from "framer-motion";
import React from "react";
import "@testing-library/jest-dom";
import { fullPermissions } from "./__mocks__/branding-permissions";
import BrandingPage from "../pages/branding";

describe.skip("Test if the Branding page is working as expected", () => {
    beforeAll(() => {
        MotionGlobalConfig.skipAnimations = true;
    });

    /**
     * Handling snapshots currently fails due to the usage of
     * framer-motion library.
     */
    it.skip("<BrandingPage /> matches snapshot", () => {
        const { container } = render(<BrandingPage />, {
            allowedScopes: fullPermissions
        });

        expect(container).toMatchSnapshot();
    });

    it("<BrandingPage /> renders properly", async () => {
        render(<BrandingPage />, {
            allowedScopes: fullPermissions
        });

        expect(screen.getByTestId("branding-page-layout")).toBeInTheDocument();

        await waitFor(() => {
            expect(screen.getByTestId("branding-preference-preview-iframe")).not.toHaveClass("loader");
        });
    });
});
