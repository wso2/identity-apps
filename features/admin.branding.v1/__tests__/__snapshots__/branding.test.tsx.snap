// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the Branding page is working as expected <BrandingPage /> matches snapshot 1`] = `
<div>
  <div
    class="layout page-layout branding-page"
    data-componentid="branding-page-layout"
    data-testid="page-layout"
  >
    <div
      class="layout-content padded"
    >
      <div
        class="page-header-wrapper"
        data-componentid="branding-page-layout-page-header"
        data-testid="page-layout-page-header"
      >
        <div
          class="page-header-inner"
        >
          <h1
            class="ui left aligned header page-header no-max-width"
            data-componentid="branding-page-layout-page-header-text-wrapper"
            data-testid="page-layout-page-header-text-wrapper"
          >
            <span
              data-componentid="branding-page-layout-page-header-title"
              data-testid="page-layout-page-header-title"
            >
              <div
                class="title-container"
              >
                <div
                  class="title-container-heading"
                >
                  extensions:develop.branding.pageHeader.title
                </div>
              </div>
            </span>
            <div
              class="sub header sub-header"
              data-componentid="branding-page-layout-page-header-sub-title"
              data-testid="page-layout-page-header-sub-title"
            >
              <div
                class="with-label"
              >
                extensions:develop.branding.pageHeader.description
              </div>
            </div>
          </h1>
        </div>
      </div>
      <div
        class="ui hidden divider"
      />
      <div
        class="ui horizontal segments basic branding-preference-tab-group"
        data-componentid="branding-preference-tabs"
      >
        <div
          class="ui basic segment"
        >
          <div
            class="tabs resource-tabs attached"
            data-componentid="branding-preference-tabs-forms"
            data-testid="resource-tabs"
            tabidentifierurlfrag=""
          >
            <div
              class="ui top attached menu"
            >
              <a
                class="active item"
              >
                extensions:develop.branding.tabs.general.label
              </a>
              <a
                class="item"
              >
                extensions:develop.branding.tabs.design.label
              </a>
              <a
                class="item"
              >
                extensions:develop.branding.tabs.advance.label
              </a>
              <a
                class="item"
              >
                console:branding.tabs.text.label
              </a>
            </div>
            <div
              class="ui bottom attached segment active tab resource-tab-pane"
              data-componentid="branding-preference-general-tab"
              data-testid="resource-tab-pane"
            >
              <div
                class="ui basic very padded segment preference-form-container"
                data-componentid="branding-preference-general-tab-content-container"
              >
                <div
                  class="ui placeholder"
                >
                  <div
                    class="line very short"
                  />
                  <div
                    class="image"
                    style="height: 38px;"
                  />
                  <div
                    class="line"
                  />
                  <div
                    class="line"
                  />
                </div>
                <div
                  class="ui placeholder"
                >
                  <div
                    class="line very short"
                  />
                  <div
                    class="image"
                    style="height: 38px;"
                  />
                  <div
                    class="line"
                  />
                  <div
                    class="line"
                  />
                </div>
                <div
                  class="ui placeholder"
                >
                  <div
                    class="line very short"
                  />
                  <div
                    class="image"
                    style="height: 38px;"
                  />
                  <div
                    class="line"
                  />
                  <div
                    class="line"
                  />
                </div>
              </div>
              <div
                class="ui bottom attached very padded segment sticky-action-container"
                data-componentid="sticky-tab-action-panel"
              >
                <button
                  arialabel="Branding preference form save button"
                  class="ui small primary disabled button"
                  data-componentid="branding-preference-general-details-form-submit-button"
                  data-testid="primary-button"
                  disabled=""
                  tabindex="-1"
                >
                  console:branding.form.actions.save
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ui basic segment preview"
        >
          <div
            class="tabs resource-tabs attached"
            data-componentid="branding-preference-tabs-preview"
            data-testid="resource-tabs"
            tabidentifierurlfrag=""
          >
            <div
              class="ui top attached menu"
            >
              <div
                class="preview-title-bar"
                index="0"
              >
                <div>
                  console:branding.tabs.preview.label
                </div>
                <div
                  class="preview-screen-selection"
                >
                  <div
                    class="field dropdown branding-preference-custom-text-screen-dropdown"
                  >
                    <label>
                      console:brandingCustomText.screenSelectDropdown.label
                    </label>
                    <div
                      aria-expanded="false"
                      arialabel="Branding text customization screen selection"
                      class="ui selection dropdown"
                      data-componentid="screen-dropdown"
                      name="screen"
                      role="listbox"
                      tabindex="0"
                    >
                      <div
                        aria-atomic="true"
                        aria-live="polite"
                        class="divider text"
                        role="alert"
                      >
                        console:brandingCustomText.screenSelectDropdown.placeholder
                      </div>
                      <i
                        aria-hidden="true"
                        class="dropdown icon"
                      />
                      <div
                        class="menu transition"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ui bottom attached segment active tab resource-tab-pane preview-tab"
              data-componentid="branding-preference-preview-tab"
              data-testid="resource-tab-pane"
            >
              <div
                class="branding-preference-preview-container"
                data-componentid="branding-preference-preview"
              >
                <div
                  class="ui placeholder branding-preference-preview-loader"
                  data-componentid="branding-preference-preview-loader"
                >
                  <div
                    class="image"
                  />
                </div>
                <div
                  class="ui iframe loading responsive branding-preference-preview-iframe"
                  data-componentid="branding-preference-preview-iframe-wrapper"
                  style="overflow: hidden; width: 0px;"
                >
                  <iframe
                    data-componentid="branding-preference-preview-iframe"
                    id="branding-preference-preview-iframe"
                    style="transform: scale(0.5555555555555556); transform-origin: 0 0; width: 0px;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
